go/modes.rst

```diff
@@ -0,0 +1,86 @@
+Build modes
+===========
+
+* `Overview`_
+* `Building static binaries`_
+* `Using the race detector`_
+
+Overview
+--------
+
+There are a few modes in which the core go rules can be run, and the selection 
+mechanism depends on the nature of the variation.
+
+Features
+~~~~~~~~
+The most common selection mechanisms used on the command line are features and 
+output groups.
+
+Features are normally off, unless you select them with :code:`--features=featurename`
+
+Available features are:
+
+* race
+
+Output groups
+~~~~~~~~~~~~~
+
+There is a default output group that is built unless you specificall select a
+different one.
+
+If you use :code:`--output_groups=groupname` then only that output group will be 
+built, if you use :code:`--output_groups=+groupname` then that output group will
+be added to the set to be built (note the +)
+
+Output groups are also often used inside rules to pick a specify output mode of
+the rules. Only outputs that are actively select are built.
+
+Available output groups are:
+
+* race
+* static
+
+Building static binaries
+------------------------
+
+You can build binaries in static linking mode using

```

Comment:
You can build statically linked binaries using:

(Maybe mention this won't work on darwin).

Reply:
Done